### PR TITLE
Converge test data usage

### PR DESCRIPTION
**Pull request by KevinMenden**
As discussed in slack and mentioned here https://github.com/nf-core/modules/issues/242, I have restructured the test data a bit.

There are now two folders, `generic` and `genomics`. The former contains any random stuff or things that we want to remove later and doesn't fit a category, while genomics contains test data for genomics experiments, sorted by organism.

The size of the test data did now shrink from about 70MB to 17 MB. There are still some unnecessary files in generic, mostly for `STAR` and `methyldackel`, which need to be removed or exchanged, or at least use only one set of data for E.coli.

The README will also need to be formatted a bit better, and we should add some guidelines on how to add tests.

### CODE REVIEW

File: .github/filters.yml
Comment (Emiller88) [inline .github/filters.yml:69]: Does this work now?

Comment (KevinMenden) [reply .github/filters.yml:69]: Ah no ... sadly it didn't. Tried around a bit and forgot about this change. But actually I think the naming makes more sense this way.
The only way around the bowtie thing is to call the tool different, so not `build`, maybe `build-index` or something

**Comment by KevinMenden** [comment]
Okay there are only 10 tests failing now, which are either conda-related or `bowtie build`tests - so that's as good as it gets.

The data structure should be good for now as well. I would really like to merge this into modules as fast as possible, because all the new modules still use the old structure and I will have to fix all of them too.
We can take care of the stuff in generic one by one afterwards, because that's usually only used by specific tools. For new modules we should really try to use the `genomics` folder wherever possible.
I will make an issue for all the data that we still need to fix.

Also let me know what else we should have in the README or just edit it :) 

ping @drpatelh

**Comment by phue** [comment]
Do we actually need the `dummy_file.txt`?  

The alternative would be to just use `file("dummy_file.txt")` to create an empty file where needed

**Comment by KevinMenden** [comment]
Yes that should work as long as we don't check whether this file exists. If it works I'm all for it!

**Comment by phue** [comment]
It is a bit strange that the `bwameth` tests don't run. They should fail because the methylated reads are gone

**Comment by phue** [comment]
@KevinMenden 
I generated (highly) artificial bisulfite reads for SARS-CoV-2 using [Sherman](https://github.com/FelixKrueger/Sherman) 
We can build indices and bamfiles from that, so we can get rid of some E.coli data.

Shall I push it here or in a separate PR afterwards?

**Comment by KevinMenden** [comment]
> It is a bit strange that the bwameth tests don't run. They should fail because the bisulfite reads are gone

Might be because I didn't change anything in that code, then no tests are run I think.

> I generated (highly) artificial bisulfite reads for SARS-CoV-2 using Sherman
We can build indices and bamfiles from that, so we can get rid of some E.coli data.
Shall I push it here or in a separate PR afterwards?

Awesome, thanks!! 
However you want, both works 👍

File: .github/filters.yml
Comment (Emiller88) [reply .github/filters.yml:77]: It was bowtie_index before, a git blame shows @drpatelh changed it to the proper name, so maybe he can weigh in.

**Comment by KevinMenden** [comment]
@drpatelh would be great if you could have a look at this :) 
no need to look at all the tests, i Just exchanged the file paths.

I would like to get this done before adding any other module.

**Comment by drpatelh** [comment]
Looks great @KevinMenden ! Minor things. Can we rename `genomic` -> `genome` in the names of the tests files? This leaves the options open for using `transcriptome` where required too so great you have factored that in. 

Still not happy about the `test_1.2.fastq.gz`, `test_1.fastq.gz` file naming convention when you have multiple test files of the same type, they aren't listed together this way when sorting by name...maybe we go for `test2_1.fastq.gz`, `test2_2.fastq.gz` instead and rename the files in the other folders accordingly too? e.g. `bed/test.2.bed` -> `bed/test2.bed`

`bed/test.genome.sizes` isn't strictly speaking a BED file so maybe we have a `genomics/sarscov2/general/` folder for these sorts of files unless you can think of a better option?

**Comment by KevinMenden** [comment]
> Can we rename genomic -> genome in the names of the tests files? This leaves the options open for using transcriptome where required too so great you have factored that in.

I took the naming convention that is used by I think NCBI or where I downloaded this, there the genome files have `genomic` in the file name.

Okay but that's the naming convention we agreed on (are rather you suggested 😜 ) and if the only reason to change it again is to make the files float together in the folder, well ... please consider that I would have to change this for every test file where there are copies, and then go through every single module because I don't know by heart which one uses which files, change the name, potentially run the tests again and adapt the md5-sums because they sometimes depend on the file name, which I often don't know beforehand. 
It would probably be a bit quicker this time because I don't have to change the whole file path but still, I will need at a few hours just to change the name again.

Basically what I'm trying to say is - I would rather not change it *again*. 😁 
Sorry for being a bit stubborn here but I already changed every single test twice, which is incredibly boring work and I am a bit terrified from doing it again. 

I'm down for moving the `test.genome.sizes` though. 😎

**Comment by drpatelh** [comment]
Ok. Don't worry, I understand. I have had to change everything a bunch of times as we have gone along too and it's.....

Let's leave `genomic` as it is then, and if and when I have the energy I will see what I can do about renaming the test files seeing as it was my suggestion and I obviously hadn't thought it through at that point 🙂 

Thanks!

**Comment by drpatelh** [comment]
PS: If you want to find files recursively in a directory containing some sort of content you can use this nugget. It has saved me alot of time and helps when renaming / removing / changing things:

```console
find ./ -type f -exec grep -H 'test_2.2.fastq.gz' {} \;
```

**Comment by KevinMenden** [comment]
Ah thanks, that's super useful indeed!!
I was just thinking that I could probably just write a simple python script that reads through all the file names and changes all the files with exact match to a certain name. That would probably work pretty easily .... 
It would still break a bunch of tests but we could see this here and then only repair those. So this way it wouldn't be that painful actually ...
Might have a look at it, maybe I'll do it after all. Or with the next PR - there will still be some cleaning up to do but at least new modules can start using the new tests data structure once this one is merged.

**Comment by KevinMenden** [comment]
Turns out I was *a bit* paranoid - forgot that this only affects a few files, because I don't have to change the normal `test.bed` files etc. Anyway wrote a little python script for it.

**Comment by KevinMenden** [comment]
By the way a transcriptome is already included --> `test_cds_from_genomic.fasta` 
That's the NCBI way of naming it.

I could change that to `test_transcriptome.fasta` and `test_genome.fasta` using the little python helper. It will probably cause some trouble with the `test.yml` files and indices but shouldn't be too bad if the name-changing is already done. So let me know what you think.
I will start compiling the VCF files for sarscov2 and also look into the remaining 'generic' files today - but this will be a separate PR.

**Comment by drpatelh** [comment]
Ah, that wasn't obvious to me at the time although it is now that you have pointed it out. Yes, renaming would be great please 😎 It is how we have have named those sorts of files in nf-core/rnaseq and how they are named on AWS iGenomes too.

File: tests/data/README.md
Comment (MaxUlysse) [inline tests/data/README.md:40]: we can gzip any gff file for this tabix test now that we have the tabix_gzip module as well

Comment (KevinMenden) [reply tests/data/README.md:40]: yes good point I forgot about that. Think it just has to be bgzip and then it works.

Comment (KevinMenden) [reply tests/data/README.md:40]: swapped it

**Comment by KevinMenden** [comment]
Shall we merge @drpatelh ? (need an approval)

**Comment by drpatelh** [comment]
Yup! Let's do it! Thanks @KevinMenden :)

**Review by drpatelh — APPROVED** [review]

**Pull request merged**
